ivadomed/loader/adaptative.py

```diff
@@ -371,100 +298,112 @@ def _load_filenames(self):
             else:
                 grp.attrs['slices'] = useful_slices
 
+            # Find contrasts
+            contrasts = set()
+            for slice_metadata in slice_seg_pair['input_metadata']:
+                contrasts.add(slice_metadata['contrast'])
             # Creating datasets and metadata
-            contrast = input_metadata['contrast']
```

Comment:
Loop across all contrasts instead of one contrast from the first input_metadata.